### DISCUSSION TITLE
Reduce size of ibdata1 file

### DISCUSSION THREAD

**Discussion by Kuepp1** [nextgenhealthcare/connect#6068] [General]
Hello,

we are using a mirth instance using mysql. Unfortunately we didn't recognise that some pdf transfer channels stored all the pdf in the database. Now the ibdata1 file is pretty large and i am thinking about a way to reduce it.

I read that there is a config paramter to use: innodb_file_per_table=1

Before i set this to the my.conf i'd thought i better ask some of you guys if there is anything else i can do or if someone has experienced problems in setting this parameter.

Maybe my approach is wrong and you have some other tips for me.

Thank you for your time!

**Comment by jonbartels** [comment]
When you do maintenance on MySQL it might help to understand the mirth table structure [with this diagram](https://mirthschema.diridium.com/). There are a lot of tables called `d_m?##` the number is the LOCAL channel ID for a particular channel. So if you see a large table for `d_mc13` then you can look at the Channels list and find Local ID 13 to see what channel it is.

I am a postgres expert but not MySQL, I hope another guru can comment with some MySQL specific advice.

1. Read about [pruning](https://docs.nextgen.com/bundle/Mirth_User_Guide_4_4_0/page/connect/connect/topics/c_Data_Pruner_Settings_Tab_connect_ug.html) and [message retention](https://docs.nextgen.com/bundle/Mirth_User_Guide_4_4_0/page/connect/connect/topics/c_Message_Pruning_Settings_connect_ug.html)
2. Find your largest tables. If you do not need the message history then TRUNCATE the `d_mc##` table for that channel.
3. Review your message storage settings. Do you need ALL the data forever?
